.github/workflows/tests.yaml

```diff
@@ -0,0 +1,28 @@
+name: Tests 
+
+on:
+  push:
+    branches: [ main, v1 ]
+  pull_request:
+    branches: [ main, v1 ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: v1
+
+    strategy:
+      matrix:
+        node-version: [16.x]
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+      - run: npm ci
+      - run: npm run build --if-present
```

Comment:
I wanted to keep this consistent with how the actual build happens so this was an explicit choice

Reply:
That makes sense. What I mean here is "do we really need a build step at all?".

IMO since this isn't going to be a module consumed by other code (for now) the build step would just be extra work rather than providing something useful. tsnode would mean we can skip it entirely and just run the ts directly, whether it's for tests or in general.

Reply:
gotcha, I just did an explicit build step because you mentioned that `ts-node` felt a bit too much like magic and preferred to keep an explicit build step here. Can totally replace with `ts-node` as well, lmk

Reply:
Yeah, I still think ts-node is a little magic, but I think I was wrong to disregard it. Especially since we don't really benefit from build artifacts at the moment, it might make it easier to run the project in general.